clients/apply/apply.go

```diff
@@ -0,0 +1,729 @@
+package apply
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influx-cli/v2/clients"
+	"github.com/influxdata/influx-cli/v2/pkg/influxid"
+	"github.com/influxdata/influx-cli/v2/pkg/template"
+)
+
+type Client struct {
+	clients.CLI
+	api.TemplatesApi
+	api.OrganizationsApi
+}
+
+type Params struct {
+	OrgId   string
+	OrgName string
+
+	StackId   string
+	Sources   []TemplateSource
+	Recursive bool
+
+	Secrets map[string]string
+	EnvVars map[string]string
+
+	Filters []ResourceFilter
+
+	Force              bool
+	OverwriteConflicts bool
+	Quiet              bool
+	RenderTableColors  bool
+	RenderTableBorders bool
+}
+
+type ResourceFilter struct {
+	Kind string
+	Name *string
+}
+
+func (c Client) Apply(ctx context.Context, params *Params) error {
+	if params.OrgId == "" && params.OrgName == "" && c.ActiveConfig.Org == "" {
+		return clients.ErrMustSpecifyOrg
+	}
+	orgID := params.OrgId
+	if orgID == "" {
+		orgName := params.OrgName
+		if orgName == "" {
+			orgName = c.ActiveConfig.Org
+		}
+		orgs, err := c.GetOrgs(ctx).Org(orgName).Execute()
+		if err != nil {
+			return fmt.Errorf("failed to get ID for org %q: %w", orgName, err)
+		}
+		if len(orgs.GetOrgs()) == 0 {
+			return fmt.Errorf("no orgs found with name %q: %w", orgName, err)
+		}
+		orgID = orgs.GetOrgs()[0].GetId()
+	}
+
+	templates, err := readTemplates(ctx, params.Sources)
+	if err != nil {
+		return err
+	}
+
+	req := api.TemplateApply{
+		DryRun:    true,
+		OrgID:     orgID,
+		Templates: templates,
+		EnvRefs:   params.EnvVars,
+		Secrets:   params.Secrets,
+		Actions:   make([]api.TemplateApplyAction, len(params.Filters)),
+	}
+	if params.StackId != "" {
+		req.StackID = &params.StackId
+	}
+	for i, f := range params.Filters {
+		req.Actions[i].Action = api.TEMPLATEAPPLYACTIONKIND_SKIP_KIND
+		if f.Name != nil {
+			req.Actions[i].Action = api.TEMPLATEAPPLYACTIONKIND_SKIP_RESOURCE
+		}
+		req.Actions[i].Properties = api.TemplateApplyActionProperties{
+			Kind:                 f.Kind,
+			ResourceTemplateName: f.Name,
+		}
+	}
+
+	// Initial dry-run to make the server summarize the template & report any missing env/secret values.
+	res, err := c.ApplyTemplate(ctx).TemplateApply(req).Execute()
+	if err != nil {
+		return fmt.Errorf("failed to check template impact: %w", err)
+	}
+
+	if len(res.Summary.MissingEnvRefs) > 0 || len(res.Summary.MissingSecrets) > 0 {
+		for _, e := range res.Summary.MissingEnvRefs {
```

Comment:
There are some places in what looks like the equivalent code in `influxdb` where there is a check to see if whatever is happening is in a TTY or not, like here https://github.com/influxdata/influxdb/blob/deaf1dfe20aa39f26bf461aa9bd66283d0037238/cmd/influx/template.go#L246 - I noticed that wasn't always duplicated here, and am curious as to why not?

Reply:
It felt redundant to the `--force` flag, which is what we use in other commands (i.e. `setup`) to bypass any interactive prompts. I guess it'll be a breaking change for any scripts out there that don't pass `--force`... will add in a check

Reply:
Added